File: Code/Components/DanpheEMR.ServerModel/WardSupplyModels/WARDInternalConsumptionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class WARDInternalConsumptionModel
    {
        [Key]
        public int ConsumptionId { get; set; }
        public int WardId { get; set; }
        public int SubStoreId { get; set; }
        public int DepartmentId { get; set; }
        public double TotalAmount { get; set; }
        public string Remark { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int?  ModifiedBy { get; set; }
        public string ConsumedBy { get; set; }

        [NotMapped]
        public List<WARDInternalConsumptionItemsModel> WardInternalConsumptionItemsList { get; set; }

    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/wardsupply/shared/ward-internal-consumption.model.ts
import {
    NgForm,
    FormGroup,
    Validators,
    FormControl,
    FormBuilder,
    ReactiveFormsModule
} from '@angular/forms'
import { WardInternalConsumptionItems } from './ward-internal-consumption-items.model';

export class WardInternalConsumption {
    public ConsumptionId: number = 0;
    public WardId: number = 0;
    public SubstoreId: number = 0;
    public DepartmentId: number = 0;
    public TotalAmount: number = 0;
    public Remark: string = '';
    public CreatedOn: Date = new Date();
    public CreatedBy: number = 0;
    public ModifiedBy: number = 0;
    public ModifiedOn: string = '';
    public ConsumedBy: string = '';
    public WardInternalConsumptionItemsList: Array<WardInternalConsumptionItems>=[];

    public InternalConsumptionValidator: FormGroup = null;
    constructor() {
        var _formBuilder = new FormBuilder();
        this.InternalConsumptionValidator = _formBuilder.group({
            'DepartmentId': ['', Validators.compose([Validators.required])],
            'Remark': ['', Validators.compose([Validators.required])],
            'ConsumedBy': ['', Validators.compose([Validators.required])]
        });
    }
    public IsDirty(fieldName): boolean {
        if (fieldName == undefined)
            return this.InternalConsumptionValidator.dirty;
        else
            return this.InternalConsumptionValidator.controls[fieldName].dirty;
    }


    public IsValid(fieldName, validator): boolean {
        if (fieldName == undefined) {
            return this.InternalConsumptionValidator.valid;
        }
        else
            return !(this.InternalConsumptionValidator.hasError(validator, fieldName));
    }
}
